ui/apps/platform/src/services/ComplianceScanConfigurationService.ts

```diff
@@ -0,0 +1,173 @@
+};
+
+export type DailySchedule = ScheduleBase & {
+    intervalType: 'DAILY';
+};
+
+export type WeeklySchedule = ScheduleBase & {
+    intervalType: 'WEEKLY';
+    daysOfWeek: { days: number[] };
+};
+
+export type MonthlySchedule = ScheduleBase & {
+    intervalType: 'MONTHLY';
+    daysOfMonth: { days: number[] };
+};
+
+export type Schedule = UnsetSchedule | DailySchedule | WeeklySchedule | MonthlySchedule;
+
+export type IntervalType = Schedule['intervalType'];
+
+type SuiteStatus = {
+    phase: string;
+    result: string;
+    errorMessage: string;
+    lastTransitionTime: string; // ISO 8601 date string
+};
+
+export type ClusterScanStatus = {
+    clusterId: string;
+    errors: string[];
+    clusterName: string;
+    suiteStatus: SuiteStatus;
+};
+
+type BaseComplianceScanConfigurationSettings = {
+    oneTimeScan: boolean;
+    profiles: string[];
+    scanSchedule: Schedule;
+    description?: string;
+};
+
+export type ComplianceScanConfiguration = {
+    id?: string;
+    scanName: string;
+    scanConfig: BaseComplianceScanConfigurationSettings;
+    clusters: string[];
+};
+
+export type ComplianceScanConfigurationStatus = {
+    id: string;
+    scanName: string;
+    scanConfig: BaseComplianceScanConfigurationSettings;
+    clusterStatus: ClusterScanStatus[];
+    createdTime: string; // ISO 8601 date string;
+    lastUpdatedTime: string; // ISO 8601 date string;
+    modifiedBy: SlimUser;
+};
+
+/*
+ * Fetches a list of scan configurations.
+ */
+export function listComplianceScanConfigurations(
+    sortOption: ApiSortOption,
+    page?: number,
+    pageSize?: number
+): Promise<ComplianceScanConfigurationStatus[]> {
+    let offset: number | undefined;
+    if (typeof page === 'number' && typeof pageSize === 'number') {
+        offset = page > 0 ? page * pageSize : 0;
+    }
+    const query = {
+        pagination: { offset, limit: pageSize, sortOption },
+    };
+    const params = qs.stringify(query, { arrayFormat: 'repeat', allowDots: true });
+    return axios
+        .get<{
+            configurations: ComplianceScanConfigurationStatus[];
+        }>(`${complianceScanConfigBaseUrl}?${params}`)
+        .then((response) => {
+            return response?.data?.configurations ?? [];
+        });
+}
+
+/*
+ * Fetches a scan configuration by ID.
+ */
+export function getComplianceScanConfiguration(
+    scanConfigId: string
+): CancellableRequest<ComplianceScanConfigurationStatus> {
+    return makeCancellableAxiosRequest((signal) =>
+        axios
+            .get<ComplianceScanConfigurationStatus>(
+                `${complianceScanConfigBaseUrl}/${scanConfigId}`,
+                {
+                    signal,
+                }
+            )
+            .then((response) => response.data)
+    );
+}
+
+/*
+ * Creates a new scan configuration or updates an existing one.
+ */
+export function saveScanConfig(
+    complianceScanConfiguration: ComplianceScanConfiguration
+): Promise<ComplianceScanConfiguration> {
```

Comment:
Brad, can you make a note for future investigation of pro and con to merge versus split put and post?

The response types of the requests differ:
https://github.com/stackrox/stackrox/blob/master/proto/api/v2/compliance_scan_configuration_service.proto#L81-L95